internal/common/util/list.go

```diff
@@ -35,3 +35,13 @@ func DeepCopyListUint32(list []uint32) []uint32 {
 	}
 	return result
 }
+
+func Filter[T any](list []T, predicate func(val T) bool) []T {
```

Comment:
Can you add some unit tests that live outside of job_test?  I’d like to see some tests around list directly?